core/ir/x86/encode.c

```diff
@@ -236,6 +236,349 @@ const reg_id_t dr_reg_fixer[] = {
     DR_REG_BND2,    DR_REG_BND3,
 };
 
+/* Maps real ISA registers to their corresponding virtual DR_ISA_REGDEPS register.
+ * Note that we map real sub-registers to their corresponding containing virtual register.
+ * Same size as dr_reg_fixer[], keep them synched.
+ */
+const reg_id_t d_r_reg_id_to_virtual[] = {
+    DR_REG_NULL,    /* DR_REG_NULL */
+    DR_REG_V0,      /* DR_REG_RAX */
+    DR_REG_V1,      /* DR_REG_RCX */
+    DR_REG_V2,      /* DR_REG_RDX */
+    DR_REG_V3,      /* DR_REG_RBX */
+    DR_REG_V4,      /* DR_REG_RSP */
+    DR_REG_V5,      /* DR_REG_RBP */
+    DR_REG_V6,      /* DR_REG_RSI */
+    DR_REG_V7,      /* DR_REG_RDI */
+    DR_REG_V8,      /* DR_REG_R8 */
+    DR_REG_V9,      /* DR_REG_R9 */
+    DR_REG_V10,     /* DR_REG_R10 */
+    DR_REG_V11,     /* DR_REG_R11 */
+    DR_REG_V12,     /* DR_REG_R12 */
+    DR_REG_V13,     /* DR_REG_R13 */
+    DR_REG_V14,     /* DR_REG_R14 */
+    DR_REG_V15,     /* DR_REG_R15 */
```

Comment:
(Would be nice to map just from the pointer-sized reg_fixer ones but it's not clear how to actually set that up...)

Reply:
I was thinking about this as well (https://github.com/DynamoRIO/dynamorio/pull/6783#discussion_r1571517251).
We could have the indexes of non-containing registers map to `DR_REG_INVALID`.
But then a user would always need to call `dr_reg_to_pointer_size()` first (otherwise they'd get a `DR_REG_INVALID`). So we might just as well save us the headache of documenting that fact and allow "users" to just pass any register and get the "correct" (w.r.t. DR_ISA_REGDEPS) mapping to virtual registers?

Reply:
If we have the full array anyway, yes.  Was hoping we could map from just the pointer-sized but we need an associative container for that: not worth it.

Reply:
Ok, leaving as is.